### PR TITLE
[OPPRO-232] Support HBM allocator

### PR DESCRIPTION
**Pull request by marin-ma**

### CODE REVIEW

**Review by jinchengchenghh — APPROVED** [review]

**Review by FelixYBW — APPROVED** [review]

**Comment by FelixYBW** [comment]
Can you fix the conflict and merge? looks it's the compile command only

File: cpp/src/CMakeLists.txt [hunk 1]
```diff
@@ -246,6 +240,20 @@ add_dependencies(spark_columnar_jni jni_proto)
 find_arrow_lib(${ARROW_LIB_NAME})
 find_arrow_lib(${GANDIVA_LIB_NAME})
 
+if(ENABLE_HBM)
```
Comment (FelixYBW) [inline cpp/src/CMakeLists.txt:243]: Do we need to install the memkind library? or build from source code?
If we need to install memkind library, let's add it to dependency list in doc.

File: cpp/src/memory/hbw_allocator_test.cc [hunk 2]
```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
```
Comment (jinchengchenghh) [inline cpp/src/memory/hbw_allocator_test.cc:6]: Can you move this one to test folder?

File: docs/Velox.md [hunk 3]
```diff
@@ -76,6 +76,20 @@ Hadoop Yarn mode is supported. Note libhdfs3 is used to read from HDFS, all its
 sudo apt install -y libiberty-dev libxml2-dev libkrb5-dev libgsasl7-dev libuuid1 uuid-dev
 ```
 
+## High-Bandwidth Memory (HBM) support
+
+Gluten supports allocating memory on HBM. This feature is optional and is disabled by default. It requires [Memkind library](http://memkind.github.io/memkind/) to be installed. Please follow memkind's [readme](https://github.com/memkind/memkind#memkind) to build and install all the dependencies and the library. 
+
+To enable this feature in Gluten, users only need to add `-Denable_hbm=ON` to the maven build command. Here's an example:
```
Comment (jinchengchenghh) [inline docs/Velox.md:83]: Please add this configuration to https://github.com/oap-project/gluten/blob/main/docs/GlutenUsage.md

**Pull request merged**